test/test_magnetic_symmetry.cpp

```diff
@@ -364,6 +364,67 @@ TEST(test_magnetic_symmetry, test_spg_get_symmetry_with_tensors_rough_symprec) {
     free(time_reversals);
 }
 
+TEST(test_magnetic_symmetry, test_spgms_get_magnetic_dataset_high_mag_symprec) {
+    // https://github.com/spglib/spglib/issues/249
+    double lattice[3][3] = {
+        {0.00000000, -5.00000000, -2.50000000},
+        {0.00000000, 0.00000000, 4.33012702},
+        {-4.05000000, 0.00000000, 0.00000000},
+    };
+    double positions[][3] = {
+        {0.50000000, 0.33333333, 0.33333333},
+        {0.50000000, 0.66666667, 0.66666667},
+        {0.00000000, 0.00000000, 0.00000000},
+        {0.00000000, 0.00000000, 0.50000000},
+        {0.00000000, 0.50000000, 0.50000000},
+        {0.00000000, 0.50000000, 0.00000000},
+    };
+    int types[] = {1, 1, 1, 2, 2, 2};
+    double tensors[] = {
+        -0.00200000, 0.00200000,  1.90000000,  0.00200000,  0.00000000,
+        1.91100000,  -0.00100000, 0.00200000,  -2.23300000, -0.00000000,
+        -0.00000000, -0.06000000, 0.00000000,  0.00000000,  -0.03200000,
+        0.00000000,  -0.00000000, -0.02900000,
+    };
+    int num_atoms = 6;
+
+    int max_size = num_atoms * 96;
+
+    double symprec = 1e-5;
+    double mag_symprec = 1e-1;  // with high mag_symprec
+
+    int i, size;
+    int equivalent_atoms[3];
+    double primitive_lattice[3][3];
+    int(*rotations)[3][3];
+    double(*translations)[3];
+    int *spin_flips;
+    int *time_reversals;
+
+    rotations = (int(*)[3][3])malloc(sizeof(int[3][3]) * max_size);
+    translations = (double(*)[3])malloc(sizeof(double[3]) * max_size);
+    spin_flips = (int *)malloc(sizeof(int *) * max_size);
+    time_reversals = (int *)malloc(sizeof(int *) * max_size);
+
+    size = spgms_get_symmetry_with_site_tensors(
+        rotations, translations, equivalent_atoms, primitive_lattice,
+        spin_flips, max_size, lattice, positions, types, tensors,
+        1 /* tensor_rank */, num_atoms, 1 /* with_time_reversal */,
+        1 /* is_axial */, symprec, -1 /* angle_tolerance */, mag_symprec);
+    ASSERT_TRUE(size > 0);
+
+    SpglibMagneticDataset *dataset;
+    dataset =
+        spgms_get_magnetic_dataset(lattice, positions, types, tensors, 1,
+                                   num_atoms, 1, symprec, -1, mag_symprec);
```

Comment:
Shouldn't you assert that `dataset == NULL`? (After you initialize to `NULL`). My understanding is that `spgms_get_magnetric_dataset` should always fail here right?

Reply:
Yes, I am adding a short comment.

Reply:
I mean not just comment, isn't this section the main part that you are testing here, and isn't it expected to always fail and return null? It seems weird that when I ran the test, it returned not-null when there is:
```c
    if (spglib_error_code == SPGLIB_SUCCESS) {
        return dataset;
    } else {
        spg_free_magnetic_dataset(dataset);
        return NULL;
    }
```
Something is fishy here

Reply:
Nvm, my last comment. :+1: to assert the null